packages/tests/server/regression/issue-4947-merged-middleware-inputs.test.ts

```diff
@@ -39,4 +39,32 @@ test('Fix #4947: standalone middlewares -- inputs are merged properly when using
       }) =>
         `valueA: ${valueA}, valueB: ${valueB}, extraProp: ${extraProp}, valueAUppercase: ${valueAUppercase}, valueBUppercase: ${valueBUppercase}`,
     );
+
+  expectTypeOf(proc._def._input_in).toEqualTypeOf<{
+    valueA: string;
+    valueB: string;
+    extraProp: string;
+  }>();
+
+  expectTypeOf(proc._def._input_out).toEqualTypeOf<{
+    valueA: string;
+    valueB: string;
+    extraProp: string;
+  }>();
```

Comment:
should we add some test when the input is transformed to make sure that also works as expected?

Reply:
I can add one to this test later today once I get off work

Reply:
I modified the test to test the types from tRPC's public types only, not private types